apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/ModuleDependency.java

```diff
@@ -146,7 +146,7 @@ public int compareTo(ModuleDependency other) {
         if (result != 0) { return result; }
         
         // do not force resolution of SPEC_VERSION_LAZY for comparisons, unnecessary
-        if (specVersion != SPEC_VERSION_LAZY || other.specVersion != SPEC_VERSION_LAZY) {
+        if (!SPEC_VERSION_LAZY.equals(specVersion) || !SPEC_VERSION_LAZY.equals(other.specVersion)) {
```

Comment:
Not a bug. SPEC_VERSION_LAZY is a class internal String that is used in one of the constructors. The comparison with `==` will detect if the class was created with this constructor.

Reply:
specVersion is variable, not a compile time constant, therefore code its correct